DataFormats/simulation/src/InteractionSampler.cxx

```diff
@@ -162,3 +162,97 @@ void InteractionSampler::setBunchFilling(const std::string& bcFillingFile)
   mBCFilling = *bc;
   delete bc;
 }
+
+// ________________________________________________
+bool NonUniformMuInteractionSampler::setBCIntensityScales(const std::vector<float>& scales_from_vector)
+{
+  // Sets the intensity scales per bunch crossing index
+  // The length of this vector needs to be compatible with the bunch filling chosen
+  mBCIntensityScales = scales_from_vector;
+
+  if (scales_from_vector.size() != mInteractingBCs.size()) {
+    LOG(error) << "Scaling factors and bunch filling scheme are not compatible. Not doing anything";
+    return false;
+  }
+
+  // since this is a scaling factor, we need to normalize the max to 1
+  float max_val = 0.;
+  for (auto v : mBCIntensityScales) {
+    max_val = std::max(max_val, v);
+  }
+  // normalize
+  for (auto& v : mBCIntensityScales) {
+    if (v < 0.) {
+      v = std::abs(v);
+    } else {
+      v /= max_val;
+    }
+  }
+  return true;
+}
+
+// ________________________________________________
+
+bool NonUniformMuInteractionSampler::setBCIntensityScales(const TH1F& hist)
+{
+  return setBCIntensityScales(determineBCIntensityScalesFromHistogram(hist));
+}
+
+std::vector<float> NonUniformMuInteractionSampler::determineBCIntensityScalesFromHistogram(const TH1F& hist)
+{
+  std::vector<float> scales;
+  // we go through the BCs and query the count from histogram
+  for (auto bc : mInteractingBCs) {
+    scales.push_back(hist.GetBinContent(bc + 1));
+  }
+  return scales;
+}
+
+int NonUniformMuInteractionSampler::getBCJump() const
+{
+  auto muFunc = [this](int bc_position) {
+    return mBCIntensityScales[bc_position % mInteractingBCs.size()] * mMuBC;
+  };
+
+  double U = gRandom->Rndm();    // uniform (0,1)
+  double T = -std::log(1.0 - U); // threshold
+  double sumMu = 0.0;
+  int offset = 0;
+  auto bcStart = mCurrBCIdx; // the current bc
+  int maxSearch = o2::constants::lhc::LHCMaxBunches;
+
+  while (offset < maxSearch) {
```

Comment:
Isn't limiting the offset to LHCMaxBunches equivalent to the assumption that the next interaction is necessarily within 1 orbit  from the current? I think there is no need for any limit at all, just check that that there is some bunch with non-0 mu.